### PR TITLE
[Communication] -Administration- Renaming the model from PhoneNumberSearch to PhoneNumberReservation

### PR DESCRIPTION
**Pull request by paolamvhz**
Renaming the model PhoneNumberSearch to PhoneNumberReservation as well as the method getSearch to getReservation

### CODE REVIEW

File: sdk/communication/azure-communication-administration/swagger/README.md [hunk 1]
```diff
@@ -44,6 +44,35 @@ These settings apply only when `--tag=phonenumber` is specified on the command l
...
 override-client-name: PhoneNumberAdminClient
 ```
+### Rename searchId to reservationId
+
+
+
+``` yaml
+
+directive:
+
```
Comment (RezaJooyandeh) [inline sdk/communication/azure-communication-administration/swagger/README.md:54]: seems that there are a few extra new lines around

**Review by RezaJooyandeh — COMMENTED** [review]
Looks very good, just a small comment. Thanks

File: sdk/communication/azure-communication-administration/README.md [hunk 2]
```diff
@@ -289,7 +289,7 @@ Duration duration = Duration.ofSeconds(1);
 String phoneNumberSearchId = "SEARCH_ID_TO_PURCHASE";
```
Comment (jbeauregardb) [inline sdk/communication/azure-communication-administration/README.md:289]: Could you also change variable names that reference the previous name of the model?

File: sdk/communication/azure-communication-administration/README.md [hunk 3]
```diff
@@ -289,7 +289,7 @@ Duration duration = Duration.ofSeconds(1);
...
 PhoneNumberClient phoneNumberClient = createPhoneNumberClient();
 
-SyncPoller<Void, Void> res = 
+SyncPoller<Void, Void> res =
     phoneNumberClient.beginPurchaseSearch(phoneNumberSearchId, duration);
```
Comment (jbeauregardb) [inline sdk/communication/azure-communication-administration/README.md:293]: Also method names

File: sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java [hunk 4]
```diff
@@ -602,10 +602,10 @@
      * Gets a search by ID.
      *
      * @param searchId ID of the search
-     * @return A {@link Mono} containing a {@link PhoneNumberSearch} representing the search.
+     * @return A {@link Mono} containing a {@link PhoneNumberReservation} representing the search.
```
Comment (jbeauregardb) [inline sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java:605]: "... representing the reservation"

File: sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberClient.java [hunk 5]
```diff
@@ -391,10 +391,10 @@ public PhoneNumberRelease getReleaseById(String releaseId) {
      * Gets a search by ID.
      *
      * @param searchId ID of the search
-     * @return A {@link PhoneNumberSearch} representing the search.
+     * @return A {@link PhoneNumberReservation} representing the search.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public PhoneNumberSearch getSearchById(String searchId) {
+    public PhoneNumberReservation getSearchById(String searchId) {
```
Comment (turalf) [inline sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberClient.java:397]: `getReservationById(String reservationId)` ?

File: sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberClient.java [hunk 6]
```diff
@@ -404,10 +404,10 @@ public PhoneNumberSearch getSearchById(String searchId) {
      * @param searchId ID of the search
      * @param context A {@link Context} representing the request context.
      * @return A {@link Response} whose {@link Response#getValue()} value returns
-     * a {@link PhoneNumberSearch} representing the search.
+     * a {@link PhoneNumberReservation} representing the search.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Response<PhoneNumberSearch> getSearchByIdWithResponse(String searchId, Context context) {
+    public Response<PhoneNumberReservation> getSearchByIdWithResponse(String searchId, Context context) {
```
Comment (turalf) [inline sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberClient.java:410]: here too; both method name and param name?

File: sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberClient.java [hunk 7]
```diff
@@ -479,26 +479,26 @@ public void cancelSearch(String searchId) {
...
+     *
      * @param options A {@link CreateSearchOptions} with the search options
-     * @param pollInterval The time our long running operation will keep on polling 
+     * @param pollInterval The time our long running operation will keep on polling
      * until it gets a result from the server
      * @return A {@link SyncPoller} object with the search result
      */
     @ServiceMethod(returns = ReturnType.COLLECTION)
-    public SyncPoller<PhoneNumberSearch, PhoneNumberSearch> beginCreateSearch(
+    public SyncPoller<PhoneNumberReservation, PhoneNumberReservation> beginCreateSearch(
```
Comment (turalf) [inline sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberClient.java:491]: `beginReservation`?

File: sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberClient.java [hunk 8]
```diff
@@ -479,26 +479,26 @@ public void cancelSearch(String searchId) {
...
         CreateSearchOptions options, Duration pollInterval) {
```
Comment (turalf) [inline sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberClient.java:492]: are we changing this to `ReservationOptions` too?

Comment (turalf) [inline sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberClient.java:492]: nit: `pollingInterval` might be better name

**Review by turalf — COMMENTED** [review]
I see there are bunch of terms with `search` in it. Should we replace all of those with `reservation`?

File: sdk/communication/azure-communication-administration/swagger/README.md (hunk 1 above)
Comment (paolamvhz) [reply sdk/communication/azure-communication-administration/swagger/README.md]: fixed

File: sdk/communication/azure-communication-administration/README.md (hunk 3 above)
Comment (paolamvhz) [reply sdk/communication/azure-communication-administration/README.md]: fixed please take a look

File: sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java (hunk 4 above)
Comment (paolamvhz) [reply sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java]: fixed please take a look

**Review by jbeauregardb — APPROVED** [review]
This looks great! Thank you very much :)

File: sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/implementation/PhoneNumberAdministrationsImpl.java [hunk 9]
```diff
@@ -2093,10 +2093,10 @@ public ReleaseResponse releasePhoneNumbers(ReleaseRequest body, Context context)
      * @return search by search id.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<PhoneNumberSearch> getSearchByIdAsync(String searchId, Context context) {
+    public Mono<PhoneNumberReservation> getSearchByIdAsync(String searchId, Context context) {
```
Comment (jbeauregardb) [inline sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/implementation/PhoneNumberAdministrationsImpl.java:2096]: "getReservationById"

File: sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberClient.java (hunk 8 above)
Comment (paolamvhz) [reply sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberClient.java]: Yes it is now changed

File: sdk/communication/azure-communication-administration/swagger/README.md [hunk 10]
```diff
@@ -45,6 +45,65 @@ input-file: https://raw.githubusercontent.com/Azure/azure-rest-api-specs/257f060
...
+### Rename searchId to reservationId in PhoneNumberSearch 
+
+``` yaml
+directive:
+  - from: swagger-document
+    where: $.definitions.PhoneNumberSearch.properties.searchId
+    transform: >
+      $["x-ms-client-name"] = "reservationId";
+```
+### Rename AddChatThreadMembersRequest to AddChatThreadMembersOptions
```
Comment (ankitarorabit) [inline sdk/communication/azure-communication-administration/swagger/README.md:66]: Is this part of this change or get added accidentally?

**Review by ankitarorabit — APPROVED** [review]

File: sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberClient.java [hunk 11]
```diff
@@ -388,134 +388,135 @@ public PhoneNumberRelease getReleaseById(String releaseId) {
...
-    public Response<CreateSearchResponse> createSearchWithResponse(CreateSearchOptions searchOptions, Context context) {
-        return phoneNumberAsyncClient.createSearchWithResponse(searchOptions, context).block();
+    public Response<CreateReservationResponse> createReservationWithResponse(
+        CreateReservationOptions reservationOptions, Context context) {
+        return phoneNumberAsyncClient.createReservationWithResponse(reservationOptions, context).block();
     }
 
     /**
-     * Gets the list of all searches
+     * Gets the list of all reservationes
```
Comment (jbeauregardb) [inline sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberClient.java:440]: reservations

File: sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberClient.java [hunk 12]
```diff
@@ -388,134 +388,135 @@ public PhoneNumberRelease getReleaseById(String releaseId) {
...
      *
-     * @return A {@link PagedIterable} of {@link PhoneNumberEntity} instances representing searches.
+     * @return A {@link PagedIterable} of {@link PhoneNumberEntity} instances representing reservationes.
```
Comment (jbeauregardb) [inline sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberClient.java:442]: same here

File: sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberClient.java [hunk 13]
```diff
@@ -388,134 +388,135 @@ public PhoneNumberRelease getReleaseById(String releaseId) {
...
     @ServiceMethod(returns = ReturnType.COLLECTION)
-    public PagedIterable<PhoneNumberEntity> listAllSearches() {
-        return new PagedIterable<>(phoneNumberAsyncClient.listAllSearches());
+    public PagedIterable<PhoneNumberEntity> listAllReservations() {
+        return new PagedIterable<>(phoneNumberAsyncClient.listAllReservations());
     }
 
     /**
-     * Gets the list of all searches
+     * Gets the list of all reservationes
```
Comment (jbeauregardb) [inline sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberClient.java:450]: same here

File: sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberClient.java [hunk 14]
```diff
@@ -388,134 +388,135 @@ public PhoneNumberRelease getReleaseById(String releaseId) {
...
      *
      * @param context A {@link Context} representing the request context.
-     * @return A {@link PagedIterable} of {@link PhoneNumberEntity} instances representing searches.
+     * @return A {@link PagedIterable} of {@link PhoneNumberEntity} instances representing reservationes.
```
Comment (jbeauregardb) [inline sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberClient.java:453]: and here

File: sdk/communication/azure-communication-administration/swagger/README.md [hunk 15]
```diff
@@ -45,6 +45,65 @@ input-file: https://raw.githubusercontent.com/Azure/azure-rest-api-specs/257f060
...
+
+``` yaml
+directive:
+    - rename-model:
+        from: CreateSearchResponse
+        to: CreateReservationResponse
+```
+
+
+
```
Comment (jbeauregardb) [inline sdk/communication/azure-communication-administration/swagger/README.md:106]: nit: remove 1 extra empty line

**Pull request merged**